src/bp/core/misc/archive.ts

```diff
@@ -0,0 +1,43 @@
+import fse from 'fs-extra'
+import glob from 'glob'
+import stream from 'stream'
+import tar from 'tar'
+import { VError } from 'verror'
+
+import { forceForwardSlashes } from './utils'
+
+export const extractArchive = async (archive: Buffer, destination: string): Promise<string[]> => {
+  if (!fse.existsSync(destination)) {
+    fse.mkdirSync(destination)
+  }
+  const buffStream = new stream.PassThrough()
+  const tarWriteStream = tar.x({ sync: true, strict: true, cwd: destination })
+
+  buffStream.end(archive)
+  buffStream.pipe(tarWriteStream)
+
+  await new Promise((resolve, reject) => {
+    tarWriteStream.on('end', resolve)
+    tarWriteStream.on('error', reject)
```

Comment:
Seems like you swallow the error here, you might want to catch it if something happens

Reply:
Exeption owuld be thrown anyway, but i'll add a try catch like the create archive so they aresimilar